src/Hit/Git/Clone.hs

```diff
@@ -35,11 +38,35 @@ git clone git@github.com:username/project-name.git
 -}
 runClone :: Text -> IO ()
 runClone txt = do
-    name <- case T.splitOn "/" txt of
-        [reponame] -> getUsername >>= \u -> pure $ u <> "/" <> reponame
-        [username, reponame] -> pure $ username <> "/" <> reponame
-        _ -> do
-            errorMessage ("Incorrect name: " <> txt <> ". Use 'repo' or 'user/repo' formats")
-            exitFailure
+    (owner, repo) <- parseOwnerRepo txt
+    let name = owner <> "/" <> repo
     let gitLink = "git@github.com:" <> name <> ".git"
     "git" ["clone", gitLink]
+    infoMessage $ " '" <> name <> "' repository is cloned to the '" <> repo <>"/' folder."
+
+{- |
+-}
+runFork :: Text -> IO ()
+runFork name = getGitHubToken >>= \case
+    Nothing -> errorMessage "Can not get GITHUB_TOKEN" >> exitFailure
+    Just auth -> do
+        (owner, repo) <- parseOwnerRepo name
+        forkExistingRepo' auth (makeName owner) (makeName repo) Nothing >>= \case
+            Left err -> do
+                errorMessage $ "Can not fork the repository: " <> name
+                errorMessage $ show err
+                exitFailure
+            Right _ -> do
+                successMessage $ " '" <> name <> "' repository is forked for your account at GitHub"
+                usr <- getUsername
+                infoMessage $ " Link: https://github.com/" <> usr <> "/" <> repo
+
+                runClone repo
```

Comment:
I think we can `setCurrentDirectory` into the cloned repo and add remote with the `upstream` name to the original repo

Reply:
This is genius! Thanks for pointing out to this :+1: